### ISSUE TITLE
Exposing ModTypeLookup, or some variation of it

### ISSUE THREAD

**Issue by Agrair** [tModLoader/tModLoader#1225]
### Description
`ModTypeLookup` is currently internal, and I believe it should be marked as public instead.

### What does this proposal attempt to solve or improve?
For library mods that add their own `ModType` content classes, it's not the cleanest to try and follow the new design pattern of `Find`/`TryFind`, since all type retrievals now redirect to `ModTypeLookup`. So, with a type `ModLiquid` that I want to be autoloaded in mods dependent on LiquidAPI, I need to make my own type dictionary. Now, it's not like that in of itself is annoying, but it becomes a problem on the front-end. I would then have to add an extension method like `Mod.GetLiquid` or `Mod.LiquidType` that reroutes to `LiquidLoader.GetLiquid`.

TL;DR: The nicety of having `ModTypeLookup` is only available to the tML base types and I would like that to change.

### Which (other) solutions should be considered?
I'm not entirely sure. Read-only access to `ModTypeLookup` is already granted through the various helper methods that already exposed. I don't know how clean it would be but there could be a similar design for write access.

**Comment by Chicken-Bones** [comment]
Make it public. Yeah it's weird to have the backing class public as well, when accessors are in `ModContent` but it's not all that different to `ContentInstance`

**Comment by Mirsario** [comment]
2a155f0 I made only Register accessible.
